### PR TITLE
 fix: parse plugin artifactItem dependencies 

### PR DESCRIPTION
**Pull request by yeikel**
Fixes https://github.com/dependabot/dependabot-core/issues/5826

### CODE REVIEW

**Comment by yeikel** [comment]
@abdulapopoola This is ready for a review 🙏

File: maven/lib/dependabot/maven/file_updater/declaration_finder.rb [hunk 1]
```diff
@@ -10,10 +10,14 @@ module Dependabot
   module Maven
     class FileUpdater
       class DeclarationFinder
-        DECLARATION_REGEX =
-          %r{<parent>.*?</parent>|<dependency>.*?</dependency>|
-             <plugin>.*?(?:<plugin>.*?</plugin>.*)?</plugin>|<extension>.*?</extension>|
-             <path>.*?</path>}mx
+        DECLARATION_REGEX = %r{
```
Comment (yeikel) [inline maven/lib/dependabot/maven/file_updater/declaration_finder.rb:13]: `artifactItem` is the only addition here. I believe that this way is more readable

**Review by abdulapopoola — APPROVED** [review]

**Comment by yeikel** [comment]
Thank you @abdulapopoola 

@jurre One more review would be appreciated. Thanks!

**Comment by yeikel** [comment]
@jeffwidman Can you help move this one forward?

**Review by jeffwidman — APPROVED** [review]
👍

**Review by abdulapopoola — APPROVED** [review]

**Pull request merged**